crates/miden-testing/src/tx_context/builder.rs

```diff
@@ -158,9 +161,19 @@ impl TransactionContextBuilder {
         self
     }
 
-    /// Override and set the [AdviceInputs]
-    pub fn advice_inputs(mut self, advice_inputs: AdviceInputs) -> Self {
-        self.advice_inputs = advice_inputs;
+    /// Extend the advice inputs with the provided [AdviceInputs] instance.
+    pub fn extend_advice_inputs(mut self, advice_inputs: AdviceInputs) -> Self {
+        self.advice_inputs.extend(advice_inputs);
+        self
+    }
```

Comment:
I think at some point we used this from the client for foreign account inputs, but we don't anymore. Since the main use-case is tests, I wonder if it should be under a feature flag. I'll double check this and tackle it as part of #980